scripts/install-cni.sh

```diff
@@ -329,13 +329,16 @@ MTU_SOURCE="<default>"
 # last field, which is the interface name. We stick to using grep to avoid
 # introducing too many new dependencies.
 
-default_nic=$(route -n | grep -E '^0\.0\.0\.0\s+\S+\s+0\.0\.0\.0' | grep -oE '\S+$')
+# In the case where there are multiple default routes (e.g. multi-networking
+# with route manager), try each one starting from the lowest metric value.
+readarray -t default_nics < <(route -n | grep -E '^0\.0\.0\.0\s+\S+\s+0\.0\.0\.0' | sort -n -k5,5 | grep -oE '\S+$')
 
 # cilium_wg0 is the interface for node-to-node encryption. If it's available /
 # in use, it has a lower MTU than the default NIC (eth0) due to encryption headers.
-for nic in cilium_wg0 "$default_nic"; do
-  if [ -f "/sys/class/net/$nic/mtu" ]; then
-    MTU=$(cat "/sys/class/net/$nic/mtu")
+for nic in cilium_wg0 "${default_nics[@]}"; do
+  mtu_file=${SYS_CLASS_NET:-/sys/class/net}/$nic/mtu
```

Comment:
Not familiar with the testing restrictions, but I take it this was introduced because during tests you are not allowed to write to `/sys/class/net`?

If my understanding is correct, let's document that.

Reply:
Is it ever possible to write to /sys/class/net anywhere?

Reply:
Ha, fair enough -- I had never tried 😝